framework/src/operators/packet_batch.rs

```diff
@@ -147,9 +149,10 @@ impl PacketBatch {
                     let ret = mbuf_free_bulk(array_ptr, len as i32);
                     self.scratch.clear();
                     if ret == 0 {
-                        Some(len)
+                        Ok(len)
                     } else {
-                        None
+                        self.clear_packets();
```

Comment:
again not sure what to do, but this seems pretty extreme.

Reply:
after convo, just going to bubble up errors and not clear.